### ISSUE TITLE
Travis build failing. Mine too.

### ISSUE THREAD

**Issue by mtnygard** [IsaacWoods/pebble#54]
Not sure if this is a problem for Pebble or the Rust nightly...

https://travis-ci.org/github/IsaacWoods/pebble/builds/682635635

I get the same failure with a local build as well.

```
mtnygard@spark:~/work/pebble/kernel$ make
cargo xbuild --target=x86_64-unknown-uefi --manifest-path efiloader/Cargo.toml
    Updating crates.io index
   Compiling core v0.0.0 (/home/mtnygard/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore)
   Compiling compiler_builtins v0.1.27
   Compiling rustc-std-workspace-core v1.99.0 (/home/mtnygard/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/tools/rustc-std-workspace-core)
error: <inline asm>:2:33: error: unknown flag
            .section .llvmbc,"e"
                                ^


error: <inline asm>:3:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^


error: aborting due to 2 previous errors

error: could not compile `rustc-std-workspace-core`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: <inline asm>:2:33: error: unknown flag
            .section .llvmbc,"e"
                                ^


error: <inline asm>:3:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^


error: <inline asm>:2:33: error: unknown flag
            .section .llvmbc,"e"
                                ^


error: <inline asm>:3:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^


error: <inline asm>:55:33: error: unknown flag
            .section .llvmbc,"e"
                                ^


error: <inline asm>:56:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^


error: aborting due to 4 previous errors

error: could not compile `compiler_builtins`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: aborting due to 2 previous errors

error: could not compile `core`.

To learn more, run the command again with --verbose.
error: `"/home/mtnygard/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo" "rustc" "-p" "alloc" "--release" "--manifest-path" "/tmp/cargo-xbuild.POPwZ5D42GUA/Cargo.toml" "--target" "x86_64-unknown-uefi" "--" "-Z" "force-unstable-if-unmarked"` failed with exit code: Some(101)
note: run with `RUST_BACKTRACE=1` for a backtrace
Makefile:8: recipe for target 'efiloader' failed
make: *** [efiloader] Error 1
```

**Comment by IsaacWoods** [comment]
Hmm, interesting, I'm wondering if this is to do with the recent [work on LLVM bitcode](https://github.com/rust-lang/rust/commit/ef89cc8f042a980e72e9d0262c267bfffd9e75fe) not interacting well with the UEFI target.

For now, you could move to an earlier nightly - I can build `master` locally with `rustc 1.45.0-nightly (fa51f810e 2020-04-29)` for example. Hope this helps!

I'll leave this issue open to track this issue, and will try and upstream a fix.

<details>
<summary>Notes about the issue:</summary>
I think the offending bit of code is likely [this](https://github.com/rust-lang/rust/commit/ef89cc8f042a980e72e9d0262c267bfffd9e75fe#diff-a3c60982b42eac1d2e3fd67f4d562ae8R871-R876) . Maybe it's as simple as checking for the UEFI target here, and emitting the Windows version?
</details>

**Comment by mtnygard** [comment]
Thank you, that definitely looks like the culprit commit. I rolled back to `nightly-2020-04-23`, which was the last complete nightly prior to the change. Was able to build successfully!

**Comment by IsaacWoods** [comment]
Thanks for the report @mtnygard!

This will hopefully be fixed by rust-lang/rust#71881
